src/app/pages/register/register.page.ts

```diff
@@ -38,21 +42,31 @@ export class RegisterPage implements OnInit {
         Validators.minLength(5),
         Validators.required
       ])),
+      name: new FormControl('', Validators.required)
     });
   }
 
   tryRegister(value){
     this.authService.registerUser(value)
-     .then(res => {
-        this.authService.loginUser(value)
-        .then(res => {
-          this.navCtrl.navigateRoot('home');
-        }, err => {
-          this.toast.present(err.message);
+      .then(res => {
+        Promise.all([
+          res.user.updateProfile({
+            displayName: value.name,
+            photoURL: 'https://user-images.githubusercontent.com/6009640/31679076-dc7581c6-b391-11e7-87fe-a8fa89793c63.png'
+          }),
+          firebase.firestore().doc('users/' + res.user.email).set({
+            'state': 'initial state',
+            'geo': [50, 19],
+          })
+        ])
+        .then(() => {
+          firebase.auth().currentUser.reload().then(() => 
+            this.navCtrl.navigateRoot('home')          
+          )
         })
-     }, err => {
-       this.toast.present(err.message)
-     })
+      }, err => {
+        console.log(err);
```

Comment:
Why did you remove toast? I think it could be useful while debugging on a real device

Reply:
soryan